packages/@react-spectrum/breadcrumbs/stories/Breadcrumbs.stories.tsx

```diff
@@ -179,3 +187,39 @@ function renderMany(props = {}) {
     </Breadcrumbs>
   );
 }
+
+function DynamicBreadcrumbs(props = {}) {
+  const defaultItems = [
+    {key: 'Folder 1', children: 'The quick brown fox jumps over'},
+    {key: 'Folder 2', children: 'My Documents'},
+    {key: 'Folder 3', children: 'Kangaroos jump high'},
+    {key: 'Folder 4', children: 'Koalas are very cute'},
+    {key: 'Folder 5', children: 'Wombat\'s noses'},
+    {key: 'Folder 6', children: 'Wattle trees'},
+    {key: 'Folder 7', children: 'April 7'}
+  ];
+  let [items, setItems] = useState(defaultItems);
+  let domRef = useRef<DOMRefValue<HTMLDivElement>>();
+  return (<>
+    <Breadcrumbs
+      ref={domRef}
+      {...props}
+      onAction={key => {
+        // Update breadcrumbs based on activated item.
+        setItems(items.slice(0, items.findIndex(item => key === item.key) + 1));
+
+        // Wait for breadcrumb items to render, then focus the current item.
+        requestAnimationFrame(() => {
```

Comment:
Would this RAF code be needed every time someone implements controlled Breadcrumbs?

Reply:
I suspect so. On action, the items get updated, and I think we need to wait a frame for them to render before restoring focus.

Reply:
Is there any way to bake this into the Breadcrumbs source code?

Reply:
This RAF is necessary for the collapsed dropdown menu use case, where the Breadcrumbs re-render. I'm not sure I can bake this into Breadcrumbs source code before the onAction event.

Reply:
I was able to refactor the focus management after onAction into the Breadcrumbs themselves. See https://github.com/adobe/react-spectrum/commit/3f2dabb48a671689f486e2cd6047f3c17e24e06d